plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1899,13 +1900,25 @@ protected void fillHostInfo(final Connection conn, final StartupRoutingCommand c
         }
     }
 
+    protected void syncPlatformAndCoresPerSocketSettings(String coresPerSocket, Map<String, String> platform) {
+        if (org.apache.commons.lang3.StringUtils.isBlank(coresPerSocket) || MapUtils.isEmpty(platform)) {
```

Comment:
add to `platform`, even if it doesn't have any elements ?

```suggestion
        if (org.apache.commons.lang3.StringUtils.isBlank(coresPerSocket) || platform = null) {
```

Reply:
: yes but `==` not `=`!

Reply:
`isEmpty` checks whether the map is null or empty

Reply:
Oh, got your point, thanks